lib/matplotlib/widgets.py

```diff
@@ -3134,6 +3277,20 @@ def extents(self, extents):
         self.set_visible(self.visible)
         self.update()
 
+    @property
+    def rotation(self):
+        """Rotation in degree in interval [0, 45]."""
+        return np.rad2deg(self._rotation)
+
+    @rotation.setter
+    def rotation(self, value):
+        # Restrict to a limited range of rotation [0, 45] to avoid changing
+        # order of handles
+        if 0 <= value and value <= 45:
+            self._rotation = np.deg2rad(value)
```

Comment:
Are there any specific problems you encountered with rotations past this range? Wondering about the differences to #19864 here, which I think had no restrictions at all. I did play around a bit changing the limits to [0, 90] and [-90, 90] and did not notice any issues; of course at those points the SW corner will become SE and so forth, so from there on some confusion might arise.
But the current limits in particular mean that rotations between 45 and 90 degrees cannot be effected at all, and with [0, 90] many configurations would still require some combination of swapping `width`, `height` and setting `rotation`.

Reply:
If it needs additional changes to ensure the rotation will work, agreed. But then for the current one a documentation entry will be needed as pointed out by @timhoffm.